src/Aspire.Hosting/Dashboard/DashboardWebApplicationHost.cs

```diff
@@ -0,0 +1,50 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Aspire.Dashboard;
+using Aspire.Hosting.Publishing;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace Aspire.Hosting.Dashboard;
+
+/// <summary>
+/// Hosts an instance of the Dashboard web application. Starts and stops it along with the parent application.
+/// </summary>
+internal sealed class DashboardWebApplicationHost : IHostedService
+{
+    // TODO move or remove this whole class when the app host stops hosting the dashboard web app
+
+    private readonly DashboardWebApplication? _dashboardWebApp;
+
+    public DashboardWebApplicationHost(
+        ILoggerFactory loggerFactory,
+        DistributedApplicationOptions options,
+        IOptions<PublishingOptions> publishingOptions)
+    {
+        // HACK: Manifest publisher check is temporary util DcpHostService is integrated with DcpPublisher.
+        if (options.DashboardEnabled && publishingOptions.Value.Publisher != "manifest")
+        {
+            var dashboardLogger = loggerFactory.CreateLogger<DashboardWebApplication>();
+
+            _dashboardWebApp = new DashboardWebApplication(dashboardLogger);
+        }
+    }
+
+    async Task IHostedService.StartAsync(CancellationToken cancellationToken)
+    {
+        if (_dashboardWebApp is not null)
```

Comment:
Logging for when it is null

Reply:
Added to the constructor which has more context on the reason.